Sources/ApodiniUtils/Swift.swift

```diff
@@ -0,0 +1,159 @@
+            insert(mergingFn(self[idx], newElement))
+        } else {
+            insert(newElement)
+        }
+    }
+}
+
+
+
+// MARK: Sequence
+
+extension Sequence {
+    /// Returns the elements of the sequence, sorted using the given key path to a comparable value
+    /// as the comparison between elements.
+    /// - Parameter keyPath: A key path to a value of an element, that is comparable.
+    /// - Returns: A sorted array.
+    public func sorted<T: Comparable>(by keyPath: KeyPath<Element, T>, ascending: Bool = true) -> [Element] {
+        let sorted = self.sorted { $0[keyPath: keyPath] < $1[keyPath: keyPath] }
+        return ascending ? sorted : sorted.reversed()
+    }
+}
+
+
+// MARK: Collection
+
+extension Collection {
+    /// Reduces the collection using the specified closure, using the first element as the initial value
+    public func reduceIntoFirst(_ transform: (inout Element, Element) throws -> Void) rethrows -> Element? {
+        guard let first = self.first else {
+            return nil
+        }
+        return try dropFirst().reduce(into: first, transform)
+    }
+    
+    /// Reduces the collection using the specified closure, using the first element as the initial value
+    public func reduceIntoFirst(_ transform: (Element, Element) throws -> Element) rethrows -> Element? {
+        guard let first = self.first else {
+            return nil
+        }
+        return try dropFirst().reduce(first, transform)
+    }
+    
+    
+    /// Count the number of elements matching `predicate`
+    public func count(where predicate: (Element) throws -> Bool) rethrows -> Int {
+        try reduce(into: 0) { $0 += try predicate($1) ? 1 : 0 }
+    }
+    
+    
+    /// Returns the first element after the specified index, which matches the predicate
+    public func first(after idx: Index, where predicate: (Element) throws -> Bool) rethrows -> Element? {
+        return try firstIndex(from: index(after: idx), where: predicate).map { self[$0] }
+    }
+    
+    
+    public func firstIndex(after idx: Index, where predicate: (Element) throws -> Bool) rethrows -> Index? {
+        return try firstIndex(from: index(after: idx), where: predicate)
+    }
+    
+    /// Returns the first index within the collection which matches a predicate, starting at `from`.
+    public func firstIndex(from idx: Index, where predicate: (Element) throws -> Bool) rethrows -> Index? {
+        guard indices.contains(idx) else {
+            return nil
+        }
+        if try predicate(self[idx]) {
+            return idx
+        } else {
+            return try firstIndex(from: index(after: idx), where: predicate)
+        }
+    }
+}
+
+
+extension RandomAccessCollection {
+    /// Safely access the element at the specified index.
+    /// - returns: the element at `idx`, if `idx` is a valid index for subscripting into the collection, otherwise `nil`.
+    public subscript(lk_safe idx: Index) -> Element? {
```

Comment:
Sorry about prefixing everything, that's a remnant from my Objective-C days...

I do generally speaking prefer to prefix all custom extensions on foreign-defined types, so that it is easier to differentiate between calls to vendor-provided methods and calls to custom functions on vendor-defined types (the underlying assumption being that my implementations probably have worse performance/efficiency than what apple ships as part of the stdlib), but in this case adding my prefix all over the Apodini code base clearly is not a desirable thing.

I'll remove all of them.

Reply:
I already guessed that this is the case. The Objective-C days ;D
Sounds good 👍